ios-trivia-game/ViewControllers/QuestionViewController.swift

```diff
@@ -11,32 +11,68 @@ import UIKit
 class QuestionViewController: UIViewController {
 
     var roomId: String?
+
+    @IBOutlet weak var questionNumber: UILabel!
+    @IBOutlet weak var questionTitle: UILabel!
+    @IBOutlet weak var answerField: UITextField!
+    @IBOutlet weak var timer: UIBarButtonItem!
+    @IBOutlet weak var submitButton: UIButton!
+    @IBOutlet weak var suggestionLabel: UILabel!
+    
+    fileprivate var countdownTimer = Timer()
+    fileprivate var timerCount = 60
+
     var question: TriviaQuestion?
     
     override func viewDidLoad() {
         super.viewDidLoad()
 
         // Do any additional setup after loading the view.
+        countdownTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(updateCounter), userInfo: nil, repeats: true)
+        timer.title = ""
         
-        // @Nari or @Zhia: Feel free to remove the hardcoded values when you get to this
-        let questionId = 1944
-        FirebaseClient.instance.getQuestionBy(questionId: questionId, complete: { (snapshot) in
-            if let questionArray = snapshot.value as? NSArray {
-                for questionData in questionArray {
-                    if let questionDict = questionData as? NSDictionary {
-                        self.question = TriviaQuestion(dictionary: questionDict)
-                        break
-                    }
-                }
-            }
-            else if let questionDict = snapshot.value as? NSDictionary {
-                self.question = TriviaQuestion(dictionary: questionDict[questionDict.allKeys.first!] as! NSDictionary)
-
+        FirebaseClient.instance.getGameBy(roomId: roomId!, complete: {(snapshot) in
+            let value = snapshot.value
+            let gameRoom = GameRoom(dictionary: value as! NSDictionary)
+            
+            // get the current question
+            let curQuestionIndex = gameRoom.current_question
+            let questions = gameRoom.questions
+            let maxQuestions = gameRoom.max_num_of_questions
+            
+            self.questionNumber.text = "Question \(curQuestionIndex! + 1)/\(maxQuestions)"
+            
+            
+            if (curQuestionIndex! >= questions.count) {
+                // prevent index-out-of-bound error
+                return
```

Comment:
So firebase does something odd here. When I increment the current_question count in the database, somehow even w/o invoking the method, the `curQuestionIndex` was updated. I kept running into an index-out-of-bounds error, and thus I'm short-circuiting the code. If I had transitioned to final score, the flow would be incorrect because the user would transition to the final score right after question. 

I didn't spend too much time debugging it and found the path of least resistance :/